### DISCUSSION TITLE
allow to use `process.terminate()` to kill process on win32

### DISCUSSION THREAD

**Discussion by trim21** [Kludex/uvicorn#2449] [Ideas]
I'm running into a issue that current implement of uvicorn reload on win32 doesn't work, it freeze on `WatchFiles detected changes in 'app.py'. Reloading`

can we add a new flag to force `process.terminate` instead of `os.kill(self.process.pid, signal.CTRL_C_EVENT)` on win32?

**Comment by Kludex** [comment]
By flag you mean a conditional?

If someone can explain the issue, and open a PR, and add a test sure.

**Reply by trim21** [reply]
a cli flag like `--reload-use-terminate`, maybe.

I can't find real reason why current impl doesn't work in my case...

**Reply by Kludex** [reply]
No. I'm not introducing arguments for no reason.

**Reply by Kludex** [reply]
If someone give me a way to reproduce it, I'm happy to help. I don't have a win32 tho.

**Reply by trim21** [reply]
looks like some framework doesn't handle signal correctly on win32…

**Comment by trim21** [comment]
OK, this is indeed not necessary, users can just start uvicorn with `watchfiles` directly (for example `watchfiles --filter python 'uvicorn server.app:app --env-file .env --no-access-log' server`), and it will just kill uvicorn process then restart it. there is no need to let uvicorn do it.
